### PR TITLE
Add support for specifying contributing annotations via KSP option

### LINKED ISSUE #24: Support declaring custom annotations via KSP option
This would avoid the need to run code gen on (or annotate) custom annotations.

### PR DESCRIPTION
**Pull request by ZacSweers**
Resolves #24

### CODE REVIEW

File: compiler/src/main/kotlin/software/amazon/lastmile/kotlin/inject/anvil/processor/MergeComponentProcessor.kt [hunk 1]
```diff
@@ -55,8 +56,30 @@ internal class MergeComponentProcessor(
     private val codeGenerator: CodeGenerator,
     override val logger: KSPLogger,
     private val contributesSubcomponentProcessor: ContributesSubcomponentProcessor,
+    options: Map<String, String>,
 ) : SymbolProcessor, ContextAware {
 
+    private val defaultContributingAnnotations = buildSet {
```
Comment (ZacSweers) [inline compiler/src/main/kotlin/software/amazon/lastmile/kotlin/inject/anvil/processor/MergeComponentProcessor.kt:62]: pulled these up opportunistically to cache them across rounds

Comment (vRallev) [reply compiler/src/main/kotlin/software/amazon/lastmile/kotlin/inject/anvil/processor/MergeComponentProcessor.kt:62]: Good!

File: compiler/src/main/kotlin/software/amazon/lastmile/kotlin/inject/anvil/processor/MergeComponentProcessor.kt [hunk 2]
```diff
@@ -173,9 +190,12 @@ internal class MergeComponentProcessor(
         resolver: Resolver,
         contributingAnnotations: Sequence<String>,
     ): Boolean {
-        return contributingAnnotations.flatMap {
-            resolver.getNewSymbolsWithAnnotation(it)
-        }.any { true }
+        return contributingAnnotations
+            .distinct() // Save duplicate lookups
```
Comment (vRallev) [inline compiler/src/main/kotlin/software/amazon/lastmile/kotlin/inject/anvil/processor/MergeComponentProcessor.kt:194]: This should be moved to line 122.

Comment (ZacSweers) [reply compiler/src/main/kotlin/software/amazon/lastmile/kotlin/inject/anvil/processor/MergeComponentProcessor.kt]: [83df1a3](https://github.com/amzn/kotlin-inject-anvil/pull/25/commits/83df1a3fd450d33328a59eb1eb4b4436599e706c)

**Pull request merged**